src/prefect/client/client.py

```diff
@@ -895,25 +899,46 @@ def set_task_run_state(
 
         Raises:
             - ClientError: if the GraphQL mutation is bad for any reason
+
+        Returns:
+            - State: the state the current task run should be considered in
         """
         mutation = {
             "mutation($state: JSON!)": {
                 with_args(
-                    "setTaskRunState",
+                    "setTaskRunStates",
                     {
                         "input": {
-                            "taskRunId": task_run_id,
-                            "version": version,
-                            "state": EnumValue("$state"),
+                            "states": [
+                                {
+                                    "taskRunId": task_run_id,
+                                    "version": version,
+                                    "state": EnumValue("$state"),
+                                }
+                            ]
                         }
                     },
-                ): {"id"}
+                ): {"states": {"id", "status", "message"}}
             }
         }
 
         serialized_state = state.serialize()
 
-        self.graphql(mutation, variables=dict(state=serialized_state))  # type: Any
+        result = self.graphql(
+            mutation, variables=dict(state=serialized_state)
+        )  # type: Any
+        state_payload = result.data.setTaskRunStates.states[0]
+        if state_payload.status == "QUEUED":
+            # note that we don't populate the 'state' attribute of the Queued MetaState yet;
+            # this is because the actual state we want to start from is the _previous_ state
+            # of this task, which is more readily available by the caller of this method
```

Comment:
I was confused by this comment until I read the caller method -- I suggest adding one line like the following to make it clear that the state will actually be populated later

```
# If appropriate, the state attribute will be provided by the function that called this method.
```